tests/test_iot/test_iot.py

```diff
@@ -756,6 +756,140 @@ def test_delete_principal_thing():
     client.delete_certificate(certificateId=cert_id)
 
 
+@mock_iot
+def test_list_thing_groups():
```

Comment:
Can you split out the various test cases in their own methods? The setup is all the same, so that can be extracted into it's own method as well.

Would make things a bit more readable

Reply:
I'll make a thing_group_tree_generator so its more flexible

Reply:
Apart form the generator, I ended up creating a test class to group all those tests together.